wc/api.py

```diff
@@ -54,51 +53,68 @@ def __get_oauth_url(self, url, method):
             consumer_key=self.consumer_key,
             consumer_secret=self.consumer_secret,
             version=self.version,
-            method=method
+            method=method,
         )
 
         return oauth.get_oauth_url()
 
-    async def request(self, method, endpoint, data, ignore_headers=False):
+    async def request(self, method, endpoint, data, params=None, ignore_headers=False):
         """ Do requests """
         url = self.__get_url(endpoint)
         auth = None
-        params = {}
+
+        if params is None:
+            # If no params have been passed we set it to be an empty dict
+            params = {}
+        elif isinstance(params, dict):
+            # Clear the consumer key and secret if they where passed as parameters.
+            params.pop("consumer_key", None)
+            params.pop("consumer_secret", None)
```

Comment:
Perhaps add to the comment that we clear them because they should only be there `self.is_ssl is True and self.query_string_auth is True` and we will take care of that later

Reply:
I will clarify the comment

Reply:
Done